compiler/backend/src/org/jetbrains/kotlin/codegen/inline/AnonymousObjectTransformer.kt

```diff
@@ -113,21 +114,14 @@ class AnonymousObjectTransformer(
             override fun visitEnd() {}
         }, ClassReader.SKIP_FRAMES)
 
-        if (!inliningContext.isInliningLambda) {
-            sourceMapper = if (debugInfo != null && !debugInfo!!.isEmpty()) {
-                SourceMapper.createFromSmap(SMAPParser.parse(debugInfo!!))
-            } else {
-                //seems we can't do any clever mapping cause we don't know any about original class name
-                IdenticalSourceMapper
-            }
-            if (sourceInfo != null && !GENERATE_SMAP) {
-                classBuilder.visitSource(sourceInfo!!, debugInfo)
-            }
+        // When regenerating objects in lambdas, we have to pass the SMAP straight through and keep
+        // the original line numbers because SMAPParser does not parse call site markers.
+        if (inliningContext.isInliningLambda) {
+            sourceMap = SMAPParser.parseOrCreateDefault(null, sourceInfo, oldObjectType.internalName, 1, 65535)
+            sourceMapper = SourceMapper(sourceMap.sourceInfo)
         } else {
-            if (sourceInfo != null) {
-                classBuilder.visitSource(sourceInfo!!, debugInfo)
-            }
-            sourceMapper = IdenticalSourceMapper
+            sourceMap = SMAPParser.parseOrCreateDefault(debugInfo, sourceInfo, oldObjectType.internalName, 1, 65535)
```

Comment:
@max-kammerer note that this is more accepting than the old code, so generating trivial SMAPs in anonymous objects in inline function is no longer necessary. If the inliner hits an object that has no SMAP, it will create one and map all line numbers to the original file.

Reply:
> is no longer necessary

Do you mean just for compatibility with old compiler?

Reply:
Well, yeah.